source/Hardpoint.cpp

```diff
@@ -218,16 +218,18 @@ void Hardpoint::Fire(Ship &ship, vector<Projectile> &projectiles, vector<Visual>
 	// offset of (.5 * velocity) and that velocity includes the velocity of the
 	// ship that fired them.
 	Point start = ship.Position() + aim.Rotate(point) - .5 * ship.Velocity();
+	Point visualStart = ship.Position() + aim.Rotate(point);
 
 	// Apply the aim and hardpoint offset.
 	aim += angle;
 	start += aim.Rotate(outfit->HardpointOffset());
+	visualStart += aim.Rotate(outfit->HardpointOffset());
```

Comment:
Why are we doing the same thing twice?

The only difference between `start` and `visualStart` is a constant offset (` - .5 * ship.Velocity()`), so we should only apply the correction when creating the visuals. Then the math is kept in sync without duplication of effort.

If it's most clear in terms of variable naming and providing context for the offset, first create what is currently termed as "visual start" but call it something like "point of origin," and then introduce the variable used for the projectile's initial drawn position with a suitable comment explaining the need for the particular offset.